src/poly.jl

```diff
@@ -0,0 +1,119 @@
+# Interface
+Base.length(poly::Union{arb_poly_struct,ArbPoly}) =
+    ccall(@libarb(arb_poly_length), Int, (Ref{arb_poly_struct},), poly)
+Base.length(poly::Union{acb_poly_struct,AcbPoly}) =
+    ccall(@libarb(acb_poly_length), Int, (Ref{acb_poly_struct},), poly)
+Base.length(series::Union{ArbSeries,AcbSeries}) = series.degree + 1
+
+# We have to define these separately so that it overrides the default
+# correctly
+degree(series::ArbSeries) = series.degree
+degree(series::AcbSeries) = series.degree
+
+Base.eltype(::Type{<:Union{ArbPoly,ArbSeries}}) = Arb
+Base.eltype(::Type{<:Union{AcbPoly,AcbSeries}}) = Acb
+
+Base.checkbounds(::Type{Bool}, poly::Union{ArbPoly,AcbPoly}, i::Integer) = i >= 0
+Base.checkbounds(::Type{Bool}, series::Union{ArbSeries,AcbSeries}, i::Integer) =
+    0 <= i <= degree(series)
+Base.checkbounds(poly::Union{ArbPoly,ArbSeries,AcbPoly,AcbSeries}, i::Integer) =
+    checkbounds(Bool, poly, i) || throw(BoundsError(poly, i))
+
+Base.@propagate_inbounds function Base.getindex(
+    poly::T,
+    i::Integer,
+) where {T<:Union{ArbPoly,ArbSeries,AcbPoly,AcbSeries}}
+    @boundscheck checkbounds(poly, i)
+    res = eltype(T)(prec = precision(poly))
+    get_coeff!(res, poly, i)
+    return res
+end
```

Comment:
I think should also add a `ref(poly:T, i)` method which returns an `ArbRef`/`AcbRef`

Reply:
This was my initial plan, but it's slightly complicated. Arb allows getting the value of any index `i >= 0`, for `i` higher than any set index it will just give the value 0. This means that while `poly[i]` will be defined for any `i >= 0` it might not correspond to an actual `arb_struct` in memory.

In practice my experience is that you don't change polynomials element-wise very often, or at least it's definitely not where most of the computational cost lies, so it's not such a big problem.

Reply:
Ok that's a fair point :)